src/plugins/mergeCells/calculations/autofill.js

```diff
@@ -0,0 +1,429 @@
+import {extend} from '../../../helpers/object';
+import {CellCoords, CellRange} from './../../../3rdparty/walkontable/src';
+import {arrayEach} from './../../../helpers/array';
+
+/**
+ * Class responsible for all of the Autofill-related operations on merged cells.
+ *
+ * @class AutofillCalculations
+ * @plugin MergeCells
+ */
+class AutofillCalculations {
+  constructor(plugin) {
+    /**
+     * Reference to the Merge Cells plugin.
+     *
+     * @type {MergeCells}
+     */
+    this.plugin = plugin;
+    /**
+     * Reference to the MergedCellsCollection class instance.
+     *
+     * @type {MergedCellsCollection}
+     */
+    this.mergedCellsCollection = this.plugin.mergedCellsCollection;
+    /**
+     * Cache of the currently processed autofill data.
+     *
+     * @private
+     * @type {Object}
+     */
+    this.currentFillData = null;
+  }
+
+  /**
+   * Correct the provided selection area, so it's not selecting only a part of a merged cell.
+   *
+   * @param {Array} selectionArea
+   */
+  correctSelectionAreaSize(selectionArea) {
+    if (selectionArea[0] === selectionArea[2] && selectionArea[1] === selectionArea[3]) {
+      let mergedCell = this.mergedCellsCollection.get(selectionArea[0], selectionArea[1]);
+
+      if (mergedCell) {
+        selectionArea[2] = selectionArea[0] + mergedCell.rowspan - 1;
+        selectionArea[3] = selectionArea[1] + mergedCell.colspan - 1;
+      }
+    }
+  }
+
+  /**
+   * Get the direction of the autofill process.
+   *
+   * @param {Array} selectionArea The selection area.
+   * @param {Array} finalArea The final area (base + drag).
+   * @return {String} `up`, `down`, `left` or `right`.
+   */
+  getDirection(selectionArea, finalArea) {
+    let direction = null;
+
+    if (finalArea[0] === selectionArea[0] && finalArea[1] === selectionArea[1] && finalArea[3] === selectionArea[3]) {
+      direction = 'down';
+
+    } else if (finalArea[2] === selectionArea[2] && finalArea[1] === selectionArea[1] && finalArea[3] === selectionArea[3]) {
+      direction = 'up';
+
+    } else if (finalArea[1] === selectionArea[1] && finalArea[2] === selectionArea[2]) {
+      direction = 'right';
+
+    } else {
+      direction = 'left';
+    }
+
+    return direction;
+  }
+
+  /**
+   * Snap the drag area to the farthest merged cell, so it won't clip any of the cell merged cells.
```

Comment:
Please change from `so it won't clip any of the cell merged cells` to `so it won't clip other merged cells` or something like that.